### PR TITLE
Add breadcrumb bar to New CloudCare

### PR DESCRIPTION
**Pull request by wpride**
screen shot 2016-08-05 at 2 52 41 pm

cross-request: https://github.com/dimagi/formplayer/pull/66

Todo:
~~Add "App" to root~~
~~Add URL to non-active tabs~~

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ FormplayerFrontend.on("before:start", function () {
 
         regions: {
             main: "#menu-region",
+            breadcrumb: "#breadcrumb-region"
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js]: Missing trailing comma. (comma-dangle)

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/menus/menu_view.js [hunk 2]
```diff
@@ -123,10 +172,27 @@ FormplayerFrontend.module("SessionNavigate.MenuList", function (MenuList, Formpl
                 currentPage: this.options.currentPage,
                 pageCount: this.options.pageCount,
                 styles: this.options.styles,
+                tiles: this.options.tiles,
+                breadcrumbs: this.options.breadcrumbs,
+                breadcrumbClick: function(index) {
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/menus/menu_view.js]: 'index' is defined but never used (no-unused-vars)

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/menus/menu_view.js [hunk 3]
```diff
@@ -123,10 +172,27 @@ FormplayerFrontend.module("SessionNavigate.MenuList", function (MenuList, Formpl
...
+                    debugger;
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/menus/menu_view.js]: Unexpected 'debugger' statement. (no-debugger)

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-/*global Marionette, Backbone, WebFormSession, Util */
+/*global Marionette, Backbone, WebFormSession, Util, Entities */
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js]: 'Entities' is defined but never used (no-unused-vars)

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/util/util.js [hunk 5]
```diff
@@ -1,51 +1,99 @@
...
+    this.clearExceptApp = function () {
+        delete this.sessionId;
+        delete this.steps;
+        delete this.page;
+        delete this.search;
+    };
+
+    this.spliceSteps = function(index) {
+        this.steps = this.steps.splice(0, index);
+    }
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/util/util.js]: Missing semicolon. (semi)

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/util/util.js [hunk 6]
```diff
@@ -75,6 +75,10 @@ Util.CloudcareUrl = function (appId, sessionId, steps, page, search) {
         this.page = null;
         this.search= null;
     };
+
+    this.spliceSteps = function(index) {
+        this.steps = this.steps.splice(0, index);
+    }
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/util/util.js]: Missing semicolon. (semi)

File: corehq/apps/cloudcare/static/cloudcare/css/formplayer.css [hunk 7]
```diff
@@ -34,15 +34,25 @@ body {
     color: #ff8400;
     text-shadow: 1px 1px 1px #000;
     font-size: 150px;
+    cursor: pointer;
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/css/formplayer.css]: yay 👉

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/menus/menu_controller.js [hunk 8]
```diff
@@ -38,6 +38,26 @@ FormplayerFrontend.module("SessionNavigate.MenuList", function (MenuList, Formpl
...
             FormplayerFrontend.regions.main.show(menuListView.render());
+
+            if (menuResponse.breadcrumbs) {
+                MenuList.Controller.showBreadcrumbs(menuResponse.breadcrumbs);
+            }
+        },
+
+        showBreadcrumbs: function (breadcrumbs) {
+            var breadcrumbsModel = [];
+            for (var i = 0; i < breadcrumbs.length; i++) {
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/menus/menu_controller.js]: you just love them for loops 😺

**Comment by benrudolph** [comment]
nothing blocking, will merge

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/menus/menu_controller.js (hunk 8 above)
Comment (wpride) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/menus/menu_controller.js]: I'm a college Java man to the 🍖

**Pull request merged**